src/OrleansGoogleUtils/Providers/Streams/GooglePubSubAdapterFactory.cs

```diff
@@ -0,0 +1,97 @@
+﻿using Microsoft.Extensions.DependencyInjection;
+using Orleans.Providers;
+using Orleans.Providers.Streams.Common;
+using Orleans.Runtime;
+using Orleans.Streams;
+using System;
+using System.Threading.Tasks;
+
+namespace Orleans.Serialization.Providers.Streams
+{
+    public class GooglePubSubAdapterFactory<TDataAdapter> : IQueueAdapterFactory
+        where TDataAdapter : IGooglePubSubDataAdapter
+    {
+        private string _projectId;
+        private string _topicId;
+        private string _deploymentId;
+        private string _providerName;
+        private int _cacheSize;
+        private int _numSubscriptions;
+        private TimeSpan? _deadline;
+        private HashRingBasedStreamQueueMapper _streamQueueMapper;
```

Comment:
@xiazen I'm trying to make it as simple as possible so 3 guys from Gitter who asked for this provider can be unblocked. This implementation is essentially the very basic queue-based Stream provider (like Azure Queue and AWS SQS). After it get in, we can refactor it to make more flexible. There are a bunch of Google's optimization that can be used. Specially in Serialization with Protobuf. But I think it is out of scope for an initial provider.

I'm just trying to get people using GCP unblocked.